### PR TITLE
Review unit test coverage for logic transaction pool - Closes #1662

### LINKED ISSUE #1662: Review unit test coverage for logic/transaction_pool
Parent #1302

Expected behavior
Ensure the test order against the implementation and add missing tests.

Actual behavior
The tests are not in order with the implementation and there are some missing tests.

Which version(s) does this affect? (Environment, OS, etc...)
1.0.0

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?
Missing test for logic `transaction_pool`
### How did I fix it?
Adding the missing tests
### How to test it?
`mocha test/unit/logic/transaction_pool.js` 
### Review checklist

* The PR solves #1662 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: test/unit/logic/transaction_pool.js [hunk 1]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+			const invalidTransaction = {
+				id: '10431411423423423L',
+				type: transactionTypes.MULTI,
+			};
+			return expect(() => {
+				transactionPool.receiveTransactions(invalidTransaction, {});
+			}).to.throw();
+		});
+
+		it('should receive transaction already processed error when the transaction is processed before', done => {
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: Test should validate that `processUnconfirmedTransaction` was called with valid parameters and when returns error or ok

File: test/unit/logic/transaction_pool.js [hunk 2]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+					done();
+				})
+			);
+		});
+	});
+
+	describe('processBundled', () => {
+		let processVerifyTransaction;
+		beforeEach(done => {
+			resetStates();
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: Full tested object should be restore with every `it`, initialization should be performed inside first `beforeEach`

File: test/unit/logic/transaction_pool.js [hunk 3]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+		});
+
+		afterEach(done => {
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransaction
+			);
+			done();
+		});
+
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: Missed scenario when `getBundledTransactionList` returns empty array

File: test/unit/logic/transaction_pool.js [hunk 4]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+				'__private.processVerifyTransaction',
+				processVerifyTransactionStub
+			);
+			transactionPool.processBundled(() => {
+				expect(logger.debug.args[0][0]).to.deep.eql(
+					'Failed to process / verify bundled transaction: 123'
+				);
+				expect(logger.debug.args[0][1]).to.deep.eql(
+					'Failed to process bundle transaction'
+				);
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: Pending validation: 
- `removeBundledTransaction` was called with transaction.id
- transaction.bundled was deleted

File: test/unit/logic/transaction_pool.js [hunk 5]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+				1,
+				'Failed to queue bundled transaction'
+			);
+			transactionPool.processBundled(() => {
+				expect(logger.debug.args[0][0]).to.deep.eql(
+					'Failed to queue bundled transaction: 123'
+				);
+				expect(logger.debug.args[0][1]).to.deep.eql(
+					'Failed to queue bundled transaction'
+				);
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: Pending validations:
- `removeBundledTransaction` was called with transaction.id
- transaction.bundled was deleted
- `removeUnconfirmedTransaction` was called with correct parameter

File: test/unit/logic/transaction_pool.js [hunk 6]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+				done();
+			});
+		});
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: Missed scenario when there is no error

File: test/unit/logic/transaction_pool.js [hunk 7]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+			undoUnconfirmedList = transactionPool.undoUnconfirmedList;
+			done();
+		});
+
+		describe('when unconfirmed lists', () => {
+			describe('is empty', () => {
+				const transactions = [];
+
+				before(done => {
+					transactionPool.getUnconfirmedTransactionList = function() {
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: stub this function to validate it was called with correct parameters

File: test/unit/logic/transaction_pool.js [hunk 8]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+					return expect(lastIds.length).to.equal(0);
+				});
+
+				describe('modules.transactions.undoUnconfirmed', () => {
+					it('should not be called', () => {
+						return expect(dummyUndoUnconfirmed.called).to.be.false;
+					});
+				});
+
+				after(resetStates);
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: No need to resetStates when beforeEach creates the object and afterEach restore sinonSandbox

File: test/unit/logic/transaction_pool.js [hunk 9]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+			});
+
+			describe('contains 1 transaction', () => {
+				describe('that is valid', () => {
+					const validTransaction = { id: 'validTx' };
+					const transactions = [validTransaction];
+
+					before(done => {
+						transactionPool.addUnconfirmedTransaction(validTransaction);
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: No need to use this function

File: test/unit/logic/transaction_pool.js [hunk 10]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+						transactionPool.getUnconfirmedTransactionList = function() {
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: Stub this function

File: test/unit/logic/transaction_pool.js [hunk 11]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+
+						it('should be called with transaction as parameter', () => {
+							return expect(dummyUndoUnconfirmed.args[0][0]).to.deep.equal(
+								validTransaction
+							);
+						});
+					});
+
+					describe('lists', () => {
+						let index;
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: unconfirmed.index, queued.index and multisignature.index are out of the scope of this unit test

File: test/unit/logic/transaction_pool.js [hunk 12]
```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
...
+						});
+						TransactionPool.__set__(
+							'modules.transactions.undoUnconfirmed',
+							dummyUndoUnconfirmed
+						);
+
+						transactionPool.addUnconfirmedTransaction(badTransaction);
+						transactionPool.getUnconfirmedTransactionList = function() {
+							return transactions;
+						};
```
Comment (LucasIsasmendi) [inline test/unit/logic/transaction_pool.js]: No need to call these functions in this unit test, everything should be stub or directly assigned

**Comment by ManuGowda** [comment]
@LucasIsasmendi I have converted the missing tests and we have all the methods coverage, as we discussed I have stubbed wherever required.

**Pull request merged**